platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -18,20 +18,74 @@
 
 import com.swirlds.merkledb.utilities.MerkleDbFileUtils;
 import java.io.IOException;
+import java.io.RandomAccessFile;
 import java.io.UncheckedIOException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.channels.FileChannel;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
+import java.util.Deque;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.CopyOnWriteArrayList;
 
-/** A direct on disk implementation of LongList */
-public class LongListDisk extends LongList {
+/**
+ *  A direct on disk implementation of LongList. This implementation creates a temporary file to store the data.
+ *  If the user provides a file, it will be used to create a copy of the in the temporary file. If the file is absent,
+ *  it will take the name of the file provided by the user and create a temporary file with this name.
+ * <p>
+ *  Unlike the "snapshot" file, the temporary files doesn't contain the header, only the body.
+ *
+ */
+public class LongListDisk extends LongList<Long> {
+
+    private static final String STORE_PREFIX = "longListDisk";
+    private static final String DEFAULT_FILE_NAME = "LongListDisk.ll";
     /** A temp byte buffer for reading and writing longs */
-    private static final ThreadLocal<ByteBuffer> TEMP_LONG_BUFFER_THREAD_LOCAL =
-            ThreadLocal.withInitial(() -> ByteBuffer.allocateDirect(Long.BYTES).order(ByteOrder.nativeOrder()));
-    /** The disk file this LongList is based on */
-    private final Path file;
+    private final ThreadLocal<ByteBuffer> tempLongBufferThreadLocal;
+
+    /** This file channel is to work with the temporary file. The field is effectively immutable, however it can't be
+     * declared final because in some cases it has to be initialized in {@link LongListDisk#readBodyFromFileChannelOnInit}
+     */
+    private FileChannel currentFileChannel;
+
+    /** A temp byte buffer for transferring data between file channels */
+    private final ThreadLocal<ByteBuffer> transferBufferThreadLocal;
+
+    /**
+     * Offsets of the chunks that are free to be used. The offsets are relative to the start of the file.
+     */
+    private final Deque<Long> freeChunks;
+
+    /**
+     * A list of all the byte buffers that are used to read and write data from the file. This is used to ensure that
+     * the buffers are properly disposed of when the list is closed.
+     */
+    private final List<ByteBuffer> bufferRegistry;
+
+    {
```

Comment:
Would this work in the constructor?  It's kind of just hanging out in the wind as it stands...

Reply:
This is an instance initializing block. It's invoked by every constructor in the class. I understand, it looks weird but it works exactly as intended :)

Reply:
I know it works properly.  I was wondering if we could move it into the constructors and cascade them (e.g. all the other constructors call the private constructor that contains this code).  It would work the same, but be less prone to error.

Reply:
There are two logical groups of constructors: 
- the ones that are working with files
- the ones that take numerical parameters

These constructors are not directly related to each other. The best thing I can do it to copy/paste this code in two constructors. I can even extract it to a method to preserve `final` modifier. Would it be better to have two copies of this block?